packages/report-server/src/reportBuilder/output/functions/matrix/matrixBuilder.ts

```diff
@@ -0,0 +1,72 @@
+import { Matrix, Row } from '../../../types';
+import { MatrixParams } from './types';
+
+export class MatrixBuilder {
+  rows: Row[];
+
+  matrixData: Matrix;
+
+  params: MatrixParams;
+
+  constructor(rows: Row[], params: MatrixParams) {
+    this.rows = rows;
+    this.params = params;
+    this.matrixData = { columns: [], rows: [] };
+  }
+
+  assertIsString(values: unknown[]): asserts values is string[] {
```

Comment:
This isn't a major thing, but personally I much prefer when functions are generic and don't rely on any class object state that they get defined outside the class itself (can still be in the same file, that's fine)

Reply:
Thanks @rohan-bes, I agree taking it out is the best practice, because we don't want the outer functions to access it.